### ISSUE TITLE
PostCSS 8.4.15+ Breaks Stylelint customSyntax resolution

### ISSUE THREAD

**Issue by johnnybenson** [stylelint/stylelint#6252]
### What steps are needed to reproduce the bug?

Yarn 1.22.19 with Workspaces
PostCSS 8.4.15+
@stylelint/postcss-css-in-js for custom syntax

Upgrading PostCSS causes Stylelint to fail to resolve the plugin in a workspace.

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
    extends: [
        'stylelint-config-recommended',
        'stylelint-config-styled-components',
    ],
    overrides: [
        {
            files: ['**/*.tsx'],
            customSyntax: '@stylelint/postcss-css-in-js',
            rules: {
                'function-no-unknown': null,
            },
        },
    ],
};
```

### How did you run Stylelint?

yarn stylelint --config ./stylelint.config.js './src/**/*.tsx'

### Which version of Stylelint are you using?

14.9.1

### What did you expect to happen?

PostCSS 8.4.14 works
```sh
$ stylelint --config ./stylelint.config.js './src/**/*.tsx'
✨  Done in 5.96s.
```

PostCSS 8.4.15 does not work
PostCSS 8.4.16 does not work

### What actually happened?

```sh
$ yarn stylelint --config ./stylelint.config.js './src/**/*.tsx'
Error: Cannot resolve custom syntax module "@stylelint/postcss-css-in-js". Check that module "@stylelint/postcss-css-in-js" is available and spelled correctly.

Caused by: Error: Cannot find module '@stylelint/postcss-css-in-js'
Require stack:
- /project/node_modules/stylelint/lib/getPostcssResult.js
- /project/node_modules/stylelint/lib/createStylelint.js
- /project/node_modules/stylelint/lib/resolveConfig.js
- /project/node_modules/stylelint/lib/printConfig.js
- /project/node_modules/stylelint/lib/cli.js
- /project/node_modules/stylelint/bin/stylelint.js
    at getCustomSyntax (/project/node_modules/stylelint/lib/getPostcssResult.js:105:11)
    at getPostcssResult (/project/node_modules/stylelint/lib/getPostcssResult.js:41:5)
    at lintSource (/project/node_modules/stylelint/lib/lintSource.js:79:20)
    at async /project/node_modules/stylelint/lib/standalone.js:228:27
    at async Promise.all (index 6)
    at async standalone (/project/node_modules/stylelint/lib/standalone.js:267:22)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Does the bug relate to non-standard syntax?

CSS-in-JS

### Proposal to fix the bug

I cross-filed the bug with PostCSS as a regression because tool chains built on tool chains should have integration tests to prevent regressions.

https://github.com/postcss/postcss/issues/1766

**Comment by JounQin** [comment]
https://antfu.me/posts/why-reproductions-are-required

Feel free to create a new issue or comment here with minimal but runnable reproduction.
